app/jobs/get_next_a_i_message_job.rb

```diff
@@ -25,12 +25,16 @@ def perform(conversation_id, assistant_id)
       @new_message.content_text = response.dig("choices", 0, "message", "content")
     end
 
-    @new_message.broadcast_replace_to @new_message.conversation, locals: { only_scroll_down_if_was_bottom: true }
+    GetNextAIMessageJob.broadcast_updated_message(@new_message)
     @new_message.save!
     puts "\nFinished GetNextAIMessageJob.perform(#{conversation_id}, #{assistant_id})" if Rails.env.development?
 
   rescue => e
     puts "Error in GetNextAIMessageJob: #{e.inspect}"
     puts e.backtrace
   end
+
+  def self.broadcast_updated_message(message)
+    message.broadcast_replace_to message.conversation, locals: { only_scroll_down_if_was_bottom: true }
+  end
```

Comment:
Moved this into a static method so tests can easily execute it. I caught a bug that tests were not excercising because of this so now we test for conversation parts streaming in.